### ISSUE TITLE
Templates: Load from external website

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#7196]
## Feature Description

**tl;dr**: the current way of bundling templates with the plugin doesn't scale. This ticket proposes offloading the templates to the wp.stories.google site.

From the beginning we'e imagined the external templates in the editor to be loaded from an external API, i.e. a template directory. Theoretically, other users could contribute to this directory as well.

However, for simplicity's sake we didn't implement such an API initially since there were only 8 such templates, instead deferring this enhancement until it's needed, and instead bundle the templates directly with the final editor bundle.

A template consists of a JSON file (**~160KB** each) plus some metadata (name, description, tags, etc.), with images being served from the wp.stories.google site. That works well.

But now there are **55** additional templates in the making, which would cause the total file size for all templates to reach **10MB**. That's a lot! That not only makes the bundling **slower**, but also makes the bundle significantly **bigger**. A bigger bundle means users' servers could struggle with unzipping larger files.

I am proposing to move the template files to the wp.stories.google site (our `static-site` branch), in the same way that we already host the template images for the exact same reasons.

### Requirements

* Just like the template images, the template JSON files need to be versioned.  
  This way, a user on an older plugin version doesn't get templates from a newer, incompatible version.
* Template metadata (name, description, tags, etc.) needs to be translatable, in a format that is easy to share with our localization team.
* Templates need to be constantly migrated to the newest data version.  
  We currently do this in CI in the plugin, but that's in a different branch than `static-site`

## Alternatives Considered

Max had the interesting suggestion of using a WP instance somewhere for hosting the templates. Where designers could simply create a template in the editor and save it, without any development work needed to get the JSON data.

This poses a few questions though about versioning, internationalization, security, caching, and probably more.

Until we have good answers for those questions I don't think it's a viable path. Hence my suggestion to start with moving the static files to wp.stories.google first.

I also see potential for a simpler version of this: once decoupling of the editor from WP progresses, it might be simpler to have a standalone version of the editor read & write template files from disk, avoiding the need for a WP instance.

## Additional Context

Firebase's built-in support for [i18n rewrites](https://firebase.google.com/docs/hosting/i18n-rewrites) might come in handy here.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

**This section is still WIP**

**Prerequisites**

* [ ] Figure out localization process

**Actual Implementation**

CDN:

1. Move template files to new `public/static/main/templates/raw` directory in the `static-site` branch, with each template in its own directory. Each template has two files: `template.json` and `metadata.json` (format of metadata file TBD)
2. A static `public/static/main/templates/index.json` file is created during the build process, containing data about all the templates.    
  **Thought:** Including metadata+actual templates here  will make this a very long file. Not great for parsing. Perhaps it should only contain metadata information about the templates and URLs to the actual template files the plugin needs to fetch separately.
2. ...

Plugin:

1. Update plugin to fetch templates from `wp.stories.google/static/main/templates`

**After**

* [ ] Update template creation docs
* [ ] Verify localization process works

#### Considerations

* Quality assurance: 
  In the current code base we have some Jest tests to ensure the correct URL format (using `__WEB_STORIES_TEMPLATE_BASE_URL__`), existence of `pageLayoutType` keys, and that templates don't contain invisible characters. We need to retain similar checks in the new location. Perhaps as a CI step before deployment.
* Pagination needed for the API? Seems unnecessary at the moment.
* This also affects page layouts, which are essentially just templates.
